### ISSUE TITLE
Enhancement: OpenTracker support in magnetico

### ISSUE THREAD

**Issue by 5zzd** [tgragnato/magnetico#361]
Have a nice day at work!

I noticed that you are actively developing the "magnetico" project.

Importing info-hashes from the specified tracker (OpenTracker) is very important to me.

If possible, add an alternative behavior for "magneticoD".

For tests, I can provide my home OpenTracker, which I use. There aren't many torrents there, but this should be enough to check.

Please, additionally read my correspondence with the author of "magnetico":

My letter to Bora:
==================================
Good day!

I am an application administrator and I'm not strong in development.
"Magneticow" works very good, however, "magneticod" works only for global DHT. I need to configure the "magneticod" to work "only for local network".

I have "OpenTracker" ( source from https://erdgeist.org/arts/software/opentracker/ ) and several "Transmission" clients running on my local network.
"Curl" can request a list of announced "Info-hashes" from "OpenTracker".
For example:
* announce URL of my "OpenTracker" is "http://ds:6969/announce".
* top10 URL"http://ds:6969/stats?mode=top10" (text and info-hash)
* top100 URL "http://ds:6969/stats?mode=top100" (text and info-hash)
* tpbs "http://ds:6969/stats?mode=tpbs&format=txt" (info-hash, seed and leech)
* statedump "http://ds:6969/stats?mode=statedump" (info-hash and text)
* scrape "http://ds:6969/scrape" (binary)

Help me set up the "magneticod" for my purposes.
How to run "magneticod" for index my "OpenTracker"?

Thanks for any help provided. Have a good day!

------------------------------------------------------------

Reply from Bora:
==================================
Hi,

No worries at all. **In fact, you reminded me to do something that I've been postponing for a long time** and I just archived the project today.
Regarding your request, I would have gladly helped if it was simple but I am afraid it isn't. I really cannot see how you can trawl a DHT of localhosts only, except that you can change the bootstrapping nodes (you need to hosts those yourself of course) and add some iptables rules to ensure that magnetico doesn't try to connect hosts outside of your local/private network. I would think it's doable, but would require lots of tuning I reckon. Hope it helps!

**Comment by tgragnato** [comment]
Hi there,

Thank you for reaching out!

`Opentracker` itself does not implement a DHT node or direct support for **_bootstrapping a DHT network_**. Instead, Opentracker functions as a traditional tracker, using the HTTP/UDP protocols to manage peer lists and facilitate the sharing process.

To bootstrap the DHT for local torrent nodes, you would typically rely on the BitTorrent clients themselves to handle DHT operations. These clients maintain lists of known DHT nodes and can join the DHT network to find peers for torrents even without a central tracker.

If you want to create a custom DHT bootstrap node for a private network, **_you might need to run a BitTorrent client in DHT-only mode_** or modify an existing client to act as a persistent DHT node, but this falls outside the typical use of Opentracker.

**Comment by 5zzd** [comment]
Hello.

**_I may be wrong_**, but the original implementation of magneticod generated a **RANDOM** info-hash, after which DHT was used to download all the rest of the information.
I hope to use the opentracker service page as a **_replacement_** for the random info-hash generator, otherwise magnetico's behavior does not require changes.

I apologize if I misled you with the original text of the appeal.

**Comment by 5zzd** [comment]
Another alternative could be to periodically read a special file that contains a list of info-hashes (format: one line - one info-hash).

Thanks in advance!

**Comment by tgragnato** [comment]
**_No problem_**, the request was quite clear. I implemented the functionality as suggested by Bora, in such a way that the amount of changes is reasonable and the deviation from standard behavior is contained.

Since I started looking at magnetico’s code, I haven’t seen anything resembling a random infohash generator. Bootstrap nodes are used to retrieve initial DHT peers; from these, the `sample_infohashes` and `get_peers` queries directly provide the `info-hashes` that those peers are downloading, as well as additional DHT peers.

Integrating directly with the `opentracker service page` would require significant changes that are incompatible with / useless to the standard mode of operation.
